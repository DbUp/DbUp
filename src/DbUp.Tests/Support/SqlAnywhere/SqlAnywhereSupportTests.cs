using System;
using ApprovalTests;
using DbUp.Builder;
using DbUp.SqlAnywhere;
using DbUp.Tests.TestInfrastructure;
using NUnit.Framework;
using Shouldly;

namespace DbUp.Tests.Support.SqlAnywhere
{
    [TestFixture]
    public class SqlAnywhereSupportTests
    {
        private const string MultilineSqlScript = @"CREATE TABLE test ( keycol int default autoincrement, datacol char(10));
                                                    CREATE TRIGGER insert_test AFTER INSERT ON test
                                                    REFERENCING NEW AS new_row
                                                    FOR EACH ROW
                                                    BEGIN
                                                    SET aicol=new_row.keycol;
                                                    END
                                                    GO

                                                    CREATE VARIABLE aicol int;
                                                    INSERT INTO test (datacol) VALUES ('mytest');
                                                    SELECT aicol FROM dummy;";

        [Test]
        public void CanUseSqlAnywhere()
        {
            var recordingDbConnection = new RecordingDbConnection(true);
            var upgrader = DeployChanges.To
                                        .SqlAnywhereDatabase("We don't care about connection string", "nor the schema name")
                                        .OverrideConnectionFactory(recordingDbConnection)
                                        .WithScript("ScriptName", MultilineSqlScript).Build();

            var result = upgrader.PerformUpgrade();

            result.Successful.ShouldBe(true);
            var commandLog = recordingDbConnection.GetCommandLog();
            try
            {
                Approvals.Verify(commandLog, Scrubbers.ScrubDates);
            }
            finally
            {
                Console.WriteLine(commandLog);
            }
        }

        [Test]
        public void Should_Configure_ScriptPreprocessors()
        {
            UpgradeConfiguration config = null;
            var builder = DeployChanges.To.SqlAnywhereDatabase("We don't care about connection string", "SchemaName").WithScriptsEmbeddedInAssembly(GetType().Assembly);

            config = ExtractConfigurationFromBuilder(builder);

            config.ScriptPreprocessors.ShouldContain(x => x.GetType() == typeof(SqlAnywhereSqlPreprocessor));
        }

        private static UpgradeConfiguration ExtractConfigurationFromBuilder(UpgradeEngineBuilder builder)
        {
            UpgradeConfiguration config = null;
            builder.Configure(c => config = c);
            builder.Build();
            return config;
        }
    }
}