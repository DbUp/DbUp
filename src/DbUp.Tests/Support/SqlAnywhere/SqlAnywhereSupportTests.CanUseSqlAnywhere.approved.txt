Open connection
Execute scalar command: select object_id('[nor the schema name].[schemaversions]')
Dispose command
Execute non query command: CREATE TABLE test ( keycol int default autoincrement, datacol char(10))
Dispose command
Execute non query command: CREATE TRIGGER insert_test AFTER INSERT ON test
                                                    REFERENCING NEW AS new_row
                                                    FOR EACH ROW
                                                    BEGIN
                                                    SET aicol=new_row.keycol;
                                                    END
Dispose command
Execute non query command: CREATE VARIABLE aicol int
Dispose command
Execute non query command: INSERT INTO test (datacol) VALUES ('mytest')
Dispose command
Execute non query command: SELECT aicol FROM dummy
Dispose command
Execute scalar command: select object_id('[nor the schema name].[schemaversions]')
Dispose command
Execute non query command: create table [nor the schema name].[schemaversions] (
	[Id] int identity not null constraint [PK_schemaversions_Id] primary key,
	[ScriptName] nvarchar(255) not null,
	[Applied] datetime not null
)
Dispose command
Create parameter
Add parameter to command: scriptName=ScriptName
Create parameter
Add parameter to command: applied=<date>
Execute non query command: insert into [nor the schema name].[schemaversions] (ScriptName, Applied) values (:scriptName, :applied)
Dispose command
Dispose connection